app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,64 @@
+class Api::V1::ReservationsController < ApplicationController
+  def index
+    reservations = Reservation.includes(:ebike).order(created_at: 'desc')
+    render json: reservations
+  end
+
+  def show
+    reservation = Reservation.find_by(id: params[:id])
+    render json: reservation
+  end
+
+  def create
+    return unless validate_time
+    return unless validate_ebike_availability
+
+    reservation = Reservation.new(reservation_params)
+    reservation.user_id = current_user.id
+
+    if reservation.save
+      render json: reservation, status: :ok
+    else
+      render json: { error: 'Reservation not created' }, status: 400
+    end
+  end
+
+  def delete
+    reservation = Reservation.find_by(id: params[:id])
+    reservation.destroy
+    render json: reservation
+  end
```

Comment:
- [ ] What happens here if the reservation is not successfully destroyed?  Kindly add some error handling as you did in the create method (line 12). 👍